packages/wallets/src/components/WalletsCarouselContent/WalletsCarouselContent.tsx

```diff
@@ -11,60 +11,76 @@ type TProps = {
-        () =>
-            walletAccountsList?.findIndex(wallet => wallet.loginid === activeWallet?.loginid) ??
-            walletsCarouselEmblaApi?.selectedScrollSnap() ??
-            0,
-        [activeWallet?.loginid, walletAccountsList, walletsCarouselEmblaApi]
-    );
-
-    const [progressBarActiveIndex, setProgressBarActiveIndex] = useState(activeWalletIndex + 1);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [activeWallet, isActiveWalletLoading, walletsCarouselEmblaApi]);
 
+    // bind to embla events
     useEffect(() => {
-        if (isCarouselInitialized) {
-            walletsCarouselEmblaApi?.scrollTo(activeWalletIndex);
-            setProgressBarActiveIndex(activeWalletIndex + 1);
+        if (!walletsCarouselEmblaApi) {
+            return;
         }
-    }, [activeWalletIndex, isCarouselInitialized, walletsCarouselEmblaApi]);
 
-    useEffect(() => {
-        walletsCarouselEmblaApi?.on('settle', () => {
+        walletsCarouselEmblaApi.on('select', () => {
             const scrollSnapIndex = walletsCarouselEmblaApi?.selectedScrollSnap();
-            const loginid = walletAccountsList?.[scrollSnapIndex]?.loginid;
-            if (activeWallet?.loginid !== loginid) {
-                switchAccount(loginid || '');
-            }
-            onWalletSettled?.(true);
+            setSelectedEmblaIndex(scrollSnapIndex);
         });
 
-        walletsCarouselEmblaApi?.on('select', () => {
-            const scrollSnapIndex = walletsCarouselEmblaApi?.selectedScrollSnap();
-            setProgressBarActiveIndex(scrollSnapIndex + 1);
-            onWalletSettled?.(false);
+        // on settle, this is only for tutorial / onboarding plugin in some other components,
+        walletsCarouselEmblaApi.on('settle', () => {
+            onWalletSettled && onWalletSettled(true);
         });
-    }, [walletsCarouselEmblaApi, switchAccount, walletAccountsList, activeWallet?.loginid, onWalletSettled]);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [walletsCarouselEmblaApi]);
 
-    const amountOfSteps = useMemo(() => walletAccountsList?.map(wallet => wallet.loginid), [walletAccountsList]);
+    // load active wallet whenever its scrolled
+    useEffect(() => {
+        const selectedLoginId = walletAccountsList?.[selectedEmblaIndex]?.loginid;
+        if (selectedLoginId) {
+            switchAccount(selectedLoginId);
+        }
+    }, [selectedEmblaIndex, switchAccount, walletAccountsList]);
+
+    // set the initial data loading flag to false once all "is loading" flags are false,
+    // as then and only then we can display all the stuff and we want to display it permanently after that
+    useEffect(() => {
+        if (!isWalletAccountsListLoading && !isActiveWalletLoading && !isCurrencyConfigLoading) {
+            setIsInitialDataLoading(false);
+        }
+    }, [isWalletAccountsListLoading, isActiveWalletLoading, isCurrencyConfigLoading]);
```

Comment:
hey @wojciech-deriv 
do we really need `useEffect` here?
what if we will calculate `isInitialDataLoading` during rerenders.
don't know maybe it can cause flickering because we are not setting `isInitialDataLoading` to true as initial value.
but if we can do smth like:
`const isInitialDataLoading = isWalletAccountsListLoading || isActiveWalletLoading || isCurrencyConfigLoading`
it would be great because we reduce rerenders it that case
🤔

Reply:
yeah, in this case its needed (at least I have not found a way to not add it)

I initially did exactly what you proposed, but it caused flickering as other hooks are still reloading when switching account :(

Reply:
I do really dislike adding any extra state, so I would love to not add it, but I need a way to know that initial data loading is done, and subsequent loads/reloads shouldn't impact it, so effectively it forces me to add extra state "const [isInitialDataLoading, setIsInitialDataLoading] = useState(true);"